### ISSUE TITLE
Path-X Benchmark

### ISSUE THREAD

**Issue by srush** [srush/annotated-s4#9]
Implement and check that the path-x results with 16k steps work.

**Comment by siddk** [comment]
Some interim notes:
- PathFinder and Long-Range Arena tasks are in general a bit hard to parse/rather disorganized. We can add them to the main repo, but it'll add some bloat.
- To reproduce original S4 results, we need to run with a batch size of 32, L = 128 * 128 = 16,384, *with some significant architectural changes* -- namely switching from LayerNorm to BatchNorm, and reorder normalization (pre-norm instead of post).
- N_Layers goes from 4 --> 6.
- Custom LR scheduler (Reduce on plateau, with fixed patience)

I'll implement the dataset parser/loader, but thinking we can skip over the finer details for purposes of the library. This will also take a looooooong time to train.
